### PR TITLE
Upgrade: Add v1.1.x pre-check script

### PR DESCRIPTION
**Pull request by bk201**
Modify for 1.1.x cluster.

### CODE REVIEW

File: pre-check/v1.1.x/check.sh [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+    current_summary=$(mktemp)
+    kubectl get bundles.fleet.cattle.io/mcc-harvester -n fleet-local -o yaml | yq '.status.summary' > $current_summary
+
+    expected_summary=$(mktemp)
+cat > $expected_summary <<EOF
+desiredReady: 1
+ready: 1
+EOF
+
+    if ! diff <(yq -P 'sort_keys(..)' $current_summary) <(yq -P 'sort_keys(..)' $expected_summary); then
```
Comment (guangbochen) [inline pre-check/v1.1.x/check.sh:43]: the diff is also doing a print, so L45 `cat $current_summary` may seem redundant.

File: pre-check/v1.1.x/check.sh [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+
+    # nodes should not be cordoned
+    nodes=$(kubectl get nodes -o yaml)
+    unschedulable=$(echo "$nodes" | yq '.items | any_c(.spec.unschedulable == true)')
+
+    if [ "$unschedulable" = "true" ]; then
+        echo "There are unschedulable nodes:"
+        echo "$nodes" | yq '.items[] | select(.spec.unschedulable == true)  | .metadata.name'
+        rm -f $node_ready_state
+    fi
```
Comment (starbops) [inline pre-check/v1.1.x/check.sh:70]: Besides `.spec.unschedulable`, do we want also to check the KubeVirt taint?

Comment (bk201) [reply pre-check/v1.1.x/check.sh:70]: I guess we can check if there is any taint. A tainted node might cause unexpected behavior.

Comment (bk201) [reply pre-check/v1.1.x/check.sh:69]: Added in https://github.com/harvester/upgrade-helpers/pull/5/commits/42e9352541b8a2f8f4398b5a69d648289b5f076e

File: pre-check/v1.1.x/check.sh (hunk 1 above)
Comment (bk201) [reply pre-check/v1.1.x/check.sh:43]: Fixed! Thanks

File: pre-check/v1.1.x/README.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    
+    All checks pass.
+    ```
+
+    If there are any failing check, please do not proceed the upgrade.
+
+
+## Other checks
+
+These are some additional checks to do.
```
[the remark this answers is not in the document]
Comment (bk201) [reply pre-check/v1.1.x/README.md]: The section is removed and v1.0.3 README is updated too. Thanks.

**Review by starbops — APPROVED** [review]
LGTM, thanks!

**Review by guangbochen — APPROVED** [review]
LGTM, thanks.

**Pull request merged**